src/server/endpoints/profile-endpoint.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const tokensDb = require('../collections/tokens-db');
+const usersDb = require('../collections/users-db');
+const videosDb = require('../collections/videos-db');
+
+function getUserProfiles(req, res) {
+  if (req.get('Authorization') === undefined) {
+    res.status(400).json({'error': 'unAuthorization'});
+    return;
+  }
+  tokensDb.getToken(req.get('Authorization'), (tokenInfos) => {
+    if (tokenInfos._id === undefined) {
+      res.status(404).json({'error': 'not found'});
+      return;
+    }
+    usersDb.findUserInfoByUsername(req.params.username, (userInfos) => {
+      if (userInfos._id === undefined) {
```

Comment:
if the connection failed you return undefined, so this 'if' will crash the application ("cannot read property _id of undefined")